src/main/java/svattask/demo/controller/PerformanceController.java

```diff
@@ -0,0 +1,28 @@
+package svattask.demo.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import svattask.demo.application.PerformanceService;
+import svattask.demo.domain.exception.PerformanceWrongListSizeException;
+import svattask.demo.dto.PerformanceResultDto;
+
+@RestController
+@RequestMapping("/performance")
+@RequiredArgsConstructor
+public class PerformanceController {
+
+    private final PerformanceService performanceService;
+
+    @GetMapping
+    public ResponseEntity<PerformanceResultDto> runPerformance(@RequestParam long listSize) {
+        if (listSize <= 2 || listSize > 1000000) {
```

Comment:
Controller is like some adapter and there shouldn't be any execution logic. Let's move this check to service layer